go/obscuronode/enclave/state.go

```diff
@@ -97,23 +97,34 @@ func emptyState() State {
 
 // Determine the new canonical L2 head and calculate the State
 // Uses cache-ing to map the Head rollup and the State to each L1Node block.
-func updateState(b *types.Block, db DB, blockResolver obscurocommon.BlockResolver) BlockState {
+func updateState(b *types.Block, db DB, blockResolver obscurocommon.BlockResolver) *BlockState {
 	// This method is called recursively in case of Re-orgs. Stop when state was calculated already.
 	val, found := db.FetchState(b.Hash())
 	if found {
 		return val
 	}
 
+	if blockResolver.HeightBlock(b) == 0 {
+		return nil
+	}
+
+	rollups := extractRollups(b, blockResolver)
+
 	// The genesis rollup is part of the canonical chain and will be included in an L1 block by the first Aggregator.
-	if b.Hash() == obscurocommon.GenesisBlock.Hash() {
+	if len(rollups) > 0 && rollups[0].Hash() == GenesisRollup.Hash() {
```

Comment:
@tudor-malene Are we confident the rollups will always be ordered such that the first rollup specifically is the genesis rollup?